packages/dd-trace/src/opentelemetry/metrics/otlp_transformer.js

```diff
@@ -0,0 +1,255 @@
+   * Transforms metrics to OTLP format based on the configured protocol.
+   * @param {Array} metrics - Array of metric data to transform
+   * @returns {Buffer} Transformed metrics in the appropriate format
+   */
+  transformMetrics (metrics) {
+    if (this.protocol === 'http/json') {
+      return this.#transformToJson(metrics)
+    }
+    return this.#transformToProtobuf(metrics)
+  }
+
+  /**
+   * Transforms metrics to protobuf format.
+   * @param {Array} metrics - Array of metrics to transform
+   * @returns {Buffer} Protobuf-encoded metrics
+   * @private
+   */
+  #transformToProtobuf (metrics) {
+    const { protoMetricsService } = getProtobufTypes()
+
+    const metricsData = {
+      resourceMetrics: [{
+        resource: this._transformResource(),
+        scopeMetrics: this.#transformScope(metrics),
+        schemaUrl: ''
+      }]
+    }
+
+    return this._serializeToProtobuf(protoMetricsService, metricsData)
+  }
+
+  /**
+   * Transforms metrics to JSON format.
+   * @param {Array} metrics - Array of metrics to transform
+   * @returns {Buffer} JSON-encoded metrics
+   * @private
+   */
+  #transformToJson (metrics) {
+    const metricsData = {
+      resourceMetrics: [{
+        resource: this._transformResource(),
+        scopeMetrics: this.#transformScope(metrics, true)
+      }]
+    }
+    return this._serializeToJson(metricsData)
+  }
+
+  /**
+   * Creates scope metrics grouped by instrumentation scope.
+   * @param {Array} metrics - Array of metrics to transform
+   * @param {boolean} isJson - Whether to format for JSON output
+   * @returns {Object[]} Array of scope metric objects
+   * @private
+   */
+  #transformScope (metrics, isJson = false) {
+    const groupedMetrics = this._groupByInstrumentationScope(metrics)
+    const scopeMetrics = []
+
+    for (const [key, metricsInScope] of groupedMetrics) {
+      const [name, version, schemaUrl] = key.split('@')
+
+      scopeMetrics.push({
+        scope: {
+          name,
+          version,
+          droppedAttributesCount: 0
+        },
+        schemaUrl,
+        metrics: metricsInScope.map(metric =>
+          isJson ? this.#transformMetricToJson(metric) : this.#transformMetric(metric)
+        )
+      })
+    }
+
+    return scopeMetrics
+  }
+
+  /**
+   * Transforms a single metric to protobuf format.
+   * @private
+   */
+  #transformMetric (metric) {
+    const result = {
+      name: metric.name,
+      description: metric.description || '',
+      unit: metric.unit || ''
+    }
+
+    if (metric.type === 'histogram') {
+      result.histogram = {
+        dataPoints: metric.data.map(dp => ({
+          attributes: this._transformAttributes(dp.attributes),
+          startTimeUnixNano: dp.startTimeUnixNano,
+          timeUnixNano: dp.timeUnixNano,
+          count: dp.count,
+          sum: dp.sum,
+          bucketCounts: dp.bucketCounts || [],
+          explicitBounds: dp.explicitBounds || [],
+          exemplars: [],
+          flags: 0,
```

Comment:
These are optional values. Here we are encoding the default values, we may use these fields the future but I can remove them to reduce the size of the payload